course_catalog/api.py

```diff
@@ -391,41 +407,45 @@ def sync_ocw_course(
     if course_json["course_id"] in blacklist:
         is_published = False
 
-    # if course run synced before, update existing Course instance
-    courserun_instance = LearningResourceRun.objects.filter(
-        platform=PlatformType.ocw.value, run_id=uid
-    ).first()
-
-    # Make sure that the data we are syncing is newer than what we already have
-    if (
-        courserun_instance
-        and last_modified <= courserun_instance.last_modified
-        and not force_overwrite
-    ):
-        log.info("Already synced. No changes found for %s", course_prefix)
-        return None
+    log.info("Digesting %s...", course_prefix)
+    course, run = digest_ocw_course(
+        course_json, last_modified, is_published, course_prefix
+    )
 
     if upload_to_s3 and is_published:
-        # Upload all course media to S3 before serializing course to ensure the existence of links
-        parser.setup_s3_uploading(
-            settings.OCW_LEARNING_COURSE_BUCKET_NAME,
-            settings.OCW_LEARNING_COURSE_ACCESS_KEY,
-            settings.OCW_LEARNING_COURSE_SECRET_ACCESS_KEY,
-            # course_prefix now has trailing slash so [-2] below is the last
-            # actual element and [-1] is an empty string
-            course_prefix.split("/")[-2],
-        )
-        if settings.OCW_UPLOAD_IMAGE_ONLY:
-            parser.upload_course_image()
-        else:
-            parser.upload_all_media_to_s3(upload_master_json=True)
-
-    log.info("Digesting %s...", course_prefix)
-    digest_ocw_course(course_json, last_modified, is_published, course_prefix)
-    return uid
+        try:
+            parser.setup_s3_uploading(
+                settings.OCW_LEARNING_COURSE_BUCKET_NAME,
+                settings.OCW_LEARNING_COURSE_ACCESS_KEY,
+                settings.OCW_LEARNING_COURSE_SECRET_ACCESS_KEY,
+                # course_prefix now has trailing slash so [-2] below is the last
+                # actual element and [-1] is an empty string
+                course_prefix.split("/")[-2],
+            )
+            if settings.OCW_UPLOAD_IMAGE_ONLY:
+                parser.upload_course_image()
+            else:
+                parser.upload_all_media_to_s3(upload_master_json=True)
+        except:  # pylint: disable=bare-except
+            log.exception(
+                (
+                    "Error encountered uploading OCW files for %s, deleting run",
+                    course_prefix,
+                )
+            )
+            run.delete()
+            raise
+    course.published = is_published or (
+        Course.objects.get(id=course.id).runs.filter(published=True).exists()
+    )
+    course.save()
+    if course.published:
+        upsert_course(course.id)
+    else:
+        delete_course(course)
 
 
-def sync_ocw_courses(*, force_overwrite, upload_to_s3):
+def sync_ocw_courses(*, course_prefixes, blacklist, force_overwrite, upload_to_s3):
     """
     Sync OCW courses to the database
 
```

Comment:
Can we add the new arguments to the docstring?

Reply:
👍